sec2_the-ruby-programming-language/2.12_final_ruby_project/1_bcrypt.rb

```diff
@@ -0,0 +1,24 @@
+require 'bcrypt'
+
+my_password = BCrypt::Password.create("my password")
+
+puts my_password
+puts my_password.version              #=> "2a"
+puts my_password.cost                 #=> 12
```

Comment:
What is `.cost`?

Reply:
This is a method by the Bcrypt library. Go find their library and. comment here what it does.

Reply:
`.cost` is the cost needed to create a hash. The higher the cost, the harder attackers guess the password but also takes longer time for the computer to find the user's password.

Reply:
Ok. So, how do you define cost in technical terms?

Reply:
cost factor